draft-ietf-taps-interface.md

```diff
@@ -737,23 +737,45 @@ early binding when required, for example with some Network Address Translator
 
 ### Using Multicast Endpoints
 
-Specifying a multicast group address on a Local Endpoint will indicate to the Transport
-Services system that the resulting connection will be used to receive multicast messages. The
-Remote Endpoint can be used to filter incoming multicast from specific senders. Such
-a Preconnection will only support calling Listen(), not Initiate(). Calling Listen()
-will cause the Transport Services system to register for receiving multicast, such
-as issuing an IGMP join {{?RFC3376}} or using MLD for IPV6 {{?RFC4604}}. Any Connections
-that are accepted from this Listener are receive-only.
+To use multicast, a Preconnection is first created with the Remote Endpoint
+specifying the multicast group and destination port number.
 
-Similarly, specifying a multicast group address on the Remote Endpoint will indicate that the
-resulting connection will be used to send multicast messages, and that the Preconnection will
-support Initiate() but not Listen(). Any Connections created this way are send-only.
+```
+RemoteSpecifier.WithIPv4MulticastGroup(233.252.0.0)
+RemoteSpecifier.WithIPv6MulticastGroup(FF0X::DB8:0:0)
+LocalSpecifier.WithPort(5353)
+```
+
+Calling Initiate() on that Preconnection creates a Connection that can be
+used to send messages to the multicast group. The Connection object that is
+created will support Send() but not Receive(). Any Connections created this
+way are send-only, and do not join the multicast group. The resulting
+Connection will have a Local Endpoint indicating the local interface to
+which the connection is bound and a Remote Endpoint indicating the
+multicast group.
```

Comment:
In sockets the semantics for sending to a destination group are identical to sending to a destination unicast address, except for 2 things I could think of:
 - TTL/Hop Limit defaults to 1 when sending to group addresses with sockets, and must be explicitly overridden with a setsockopt to get a longer limit.  (The value 1 is the most appropriate default for L2 discovery use cases of multicast, but typically causes a few hours of debugging for bulk data delivery use cases that want the traffic to get routed for multiple hops.)
 - The destination MAC address for ethernet has a static mapping that's implied for multicast destination addresses.

I guess I'm OK with using WithIPxMulticastGroup for sending, though I think for sending it's much less helpful than it is for receiving (as compared to just setting the RemoteSpecifier WithIPxAddress for the destination group address when sending).  Also, I think this might be more confusing than helpful if we don't also use another different thing for broadcast addresses, which are different from both multicast and unicast, but like multicast, also not different in ways that are usually very important ways for senders to distinguish.

So I tend to think it might make more sense to just use WithIPxAddress to set the group destination address on the RemoteSpecifier when sending, and leave WithIPxMulticastGroup (including an ASM/SSM distinguisher) to ask for the appropriate join operation for receiving, but my opinion on this point is not strong.